.github/workflows/format.yml

```diff
@@ -0,0 +1,30 @@
+name: Format suggestions
+
+on:
+  pull_request:
+
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: julia-actions/setup-julia@19bd4db8c427e07d56602f1c432e1a26a55d6735 # v1.8.2
+        with:
+          version: 1
+      - shell: julia --color=yes {0}
+      - run: |
+          using Pkg
+          Pkg.add(PackageSpec(name="JuliaFormatter", version="1.0.10"))
+          using JuliaFormatter
+          format(".", MinimalStyle(); normalize_line_endings = "unix")
+      - uses: reviewdog/action-suggester@v1
```

Comment:
This action requires a token with write access. Therefore, it won't work on PRs made from forks.

Instead of using `reviewdog` to make inline review suggestions, we should simply fail if there are formatting issues (similar to how the existing whitespace check does). To make it easier for the PR author to identify formatting issues, before we exit, we should print the list of file names and line numbers where formatting issues are present (which is also what the existing whitespace check does).

Reply:
>The reviewdog/action-suggester action requires a token with write access. Therefore, it won't work on PRs made from forks.

I see, how does pre-commit.CI does it then? there must be a way, because pre-commit CI can push to fork PRs just fine

Reply:
we can use https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork ?

Reply:
`reviewdog/action-suggester` is a GitHub Action.

`pre-commit.ci` is a GitHub App. In order to use `pre-commit.ci`, you have to install their GitHub App on your repository and give them full write access to your entire repository. This enables them to do things such as pushing to PR branches from forks. However, it is also a big security risk - if the `pre-commit.CI` GitHub App is compromised, then your repository is compromised.

As a general rule, we want to minimize the number of GitHub Apps that we have installed on the `JuliaLang/julia` repository. In particular, we definitely want to minimize the number of GitHub Apps _with write access to the repository_ that we have installed on the `JuliaLang/julia` repository.

Reply:
while that makes sense, https://github.com/carbon-language/carbon-lang/blob/trunk/.pre-commit-config.yaml I don't it's uncommon to use pre-commit.ci, after all, you can see all the changes it made to the PR

Reply:
Every project needs to assess the pros and cons (and security risks) in the context of their project and their resources.

Google has a much larger team of CI and security people than we do, so perhaps the risk/benefit calculation looks different for them.

Reply:
how is commits made by CI any different from "a new github user making PR"

fundamentally:
- CI is not going to touch more lines than the PR is already touching
- CI commits are isolated